include/ocpp/v201/charge_point.hpp

```diff
@@ -166,6 +161,21 @@ struct Callbacks {
     /// \brief Callback function that can be used to handle arbitrary data transfers for all vendorId and
     /// messageId
     std::optional<std::function<DataTransferResponse(const DataTransferRequest& request)>> data_transfer_callback;
+
+    /* Callbacks for networking */
+    /// \brief register a \p callback that is called when the websocket is connected successfully
+    std::function < void(std::optional<NetworkConnectionProfile> network_profile)> websocket_connected_callback;
+
+    /// \brief register a \p callback that is called when the websocket connection is disconnected
+    std::function<void()>websocket_disconnected_callback;
+
+    // callback is called when receiving a SetNetworkProfile.req from the CSMS
+    std::optional<std::function<SetNetworkProfileStatusEnum(
+        const int32_t configuration_slot, const NetworkConnectionProfile& network_connection_profile)>>
+        validate_network_profile_callback;
+    std::optional<std::function<bool(const NetworkConnectionProfile& network_connection_profile,
```

Comment:
Why don't you return the promise in this callback? 

And I think you should return a future, not the promise.